feat: add CODER_OIDC_IGNORE_EMAIL_VERIFIED config knob (#5165)
coder/coder: docs/admin/auth.md

> When a new user is created, the `preferred_username` claim becomes the username. If this claim is empty, the email address will be stripped of the domain, and become the username (e.g. `example@coder.com` becomes `example`).

If your OpenID Connect provider requires client TLS certificates for authentication, you can configure them like so:

```console
CODER_TLS_CLIENT_CERT_FILE=/path/to/cert.pem
CODER_TLS_CLIENT_KEY_FILE=/path/to/key.pem
```

Coder requires all OIDC email addresses to be verified by default. If the `email_verified` claim is present in the token response from the identity provider, Coder will validate that its value is `true`.

> If you need to modify this behaviour, you can disable this behaviour like so:

Reviewer: This part felt a bit repetitive, spiffed it up a bit and US-Englishified it 😅

```suggestion
If needed, you can disable this behavior with the following setting:
```
